shell/security/prismaci/twist-scan.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+#
+# This script scans docker images for vulnerabilities, generates local
+# report and also uploads the report to Prisma Cloud. The script expects
+# special credentials - those are set by prismacloud-credentials circleci context.
+#
+
+## Add some safeties for external dependencies when running
+if [[ -z $CIRCLECI ]]; then
+  echo "I can only run in CircleCI"
+  exit 1
+fi
+if ! command -v curl >/dev/null 2>&1; then 
+  echo "We need an executable called curl in the \$PATH in order to execute"
+  exit 1
+fi
+if ! command -v jq >/dev/null 2>&1; then 
+  echo "We need an executable called curl in the \$PATH in order to execute"
+  exit 1
+fi
+if [[ -z $PC_CONSOLE_URL ]]; then
+  echo "Need an environment variable called PC_CONSOLE_URL"
+  exit 1
+fi
+if [[ -z $PC_ACCESS_KEY ]]; then
+  echo "Need an environment variable called PC_ACCESS_KEY"
+  exit 1
+fi
+if [[ -z $PC_SECRET_KEY ]]; then
+  echo "Need an environment variable called PC_SECRET_KEY"
+  exit 1
+fi
+
+## /Add some safeties for external dependencies when running
+
+PC_CLOUD_TOKEN_JSON=/tmp/token.json
+info "setting up prismacloud auth"
+curl -s -S -o ${PC_CLOUD_TOKEN_JSON} -X POST --user-agent "CircleCI/Getoutreach/2.2.0" \
+  -H "Content-type: application/json" \
+  -d '{"username": "'"${PC_ACCESS_KEY}"'", "password": "'"${PC_SECRET_KEY}"'"}' \
+  "${PC_CONSOLE_URL}/api/v1/authenticate"
+
+PC_CLOUD_TOKEN=$(jq -r -c '.token' <${PC_CLOUD_TOKEN_JSON})
+
+PC_CLOUD_TWISTCLI=/tmp/twistcli
+info "downloading twistcli from prismacloud"
+curl -L --header "authorization: Bearer ${PC_CLOUD_TOKEN}" \
+  "${PC_CONSOLE_URL}/api/v1/util/twistcli" -o ${PC_CLOUD_TWISTCLI}
+chmod a+x ${PC_CLOUD_TWISTCLI}
+
+
+if [[ -z $DOCKER_HOST ]]; then
+    # trying the default unix one
+    TWISTCLI_DOCKER_HOST=/var/run/docker.sock
+else
+    TWISTCLI_DOCKER_HOST=$(sed -e 's/tcp/https/' <<< "${DOCKER_HOST}")
+fi
+
+${PC_CLOUD_TWISTCLI} images scan --token "${PC_CLOUD_TOKEN}" \
+  --ci --details --address "${PC_CONSOLE_URL}" \
+  --custom-labels \
+  --docker-address "${TWISTCLI_DOCKER_HOST}" \
+  --output-file "${TEST_RESULTS}/image_scan.json" \
+  "$@"
```

Comment:
This would not work in this form inside the docker executor in circleci, I am not sure of the behavior in the machine executor.

There are env variables that docker client uses by convention in order to identify DOCKER_HOST and certificate materials for the docker client to use in order to validate a TLS'ed connection via dockerd.  If DOCKER_HOST is not the empty string, the twistcli is going to need the  `--docker-tlscert`, `--docker-tlscacert`, and `--docker-tlskey` flags  to be set.

Reply:
Bootstrap is going to run this script on a bare VM, not inside docker. Since docker address is going to be pure unix sock, TLS certs are no longer needed. BTW - this PR has been fully tested on a bootstraptestservice - and it prodused scan results :).

Reply:
great, then I would suggest either removing the test for DOCKER_HOST, or making that test fail if DOCKER_HOST is not the empty string. As is, this form of the twistcli command would be non-functional if DOCKER_HOST is not the empty string.

Reply:
updated